### DISCUSSION TITLE
Is there a way to prevent installing specific packages?

### DISCUSSION THREAD

**Discussion by harrisoff** [npm/feedback#513] [CLI] [answered]
For example, the project will need a time dependency in the future, at that time I want the maintainers (probably not me) to use `dayjs` instead of `moment`.

We could write some rules in documents, but sometimes we could not make sure everyone will read the docs carefully.

So is it possible to add some restrictions to the installing progress?

I imagine the config is something like this:

```jsonc
{
  // ...
  "blackList": {
    "moment": {
      // all of the options are optional
      // "message": "use dayjs instead",
      // "version": "*"
    }
  }
}
```

When user runs `npm install moment`, the install progress will break and an error will be thrown.

**Comment by MylesBorins** [comment, marked as answer]
Today there is no way to do this.

If you would like to propose a specific way to accomplish this behavior, as you have identified above, the appropriate next step would be to open an issue over in the [npm rfc repo](https://github.com/npm/rfcs]

**Reply by harrisoff** [reply]
thanks
